### DISCUSSION TITLE
Premake5 - beta3

### DISCUSSION THREAD

**Discussion by Carluccio** [premake/premake-core#2326] [General]
windows 10.
premake5 - beta3.
mingw-64.

I have a file-rc.
With gmake no problem. With gmake2 premake5 -beta3 it crashes because it does not find and does not create the obj/..... folders.
If I create them first and manually, no problem, it finds them and writes the.res file and the object files to them.

**Comment by nickclark2016** [comment]
Would it be possible for you to upload a minimum reproducing sample?

@Jarod42 can we get an integration test in your samples project to have verification against this bug?

**Reply by Jarod42** [reply]
Checking rc was in my TODO list.
Issue I have is to check that rc file is really compiled and not simply ignored.
(I accept PRs ;-) )

**Reply by Jarod42** [reply]
Just added project-rc (in branch), and I don't spot the issue (I generated also regular objs, that might explain why).
Non-windows fails as windres is not found for gmake/gmake2 (and ninja)

**Reply by Jarod42** [reply]
When reading generated makefiles,
- for gmake: there is an explicit dependency between `$(OBJECTS)` and `$(OBJDIR)`, but nothing for `$(RESOURCES)` (but build "after" (but possibly at the same time) `$(OBJECTS)`.
- for gmake2: It is similar with intermediate `prebuild`.

As I understand both are missing dependencies.
`make -j1` should work in both case though
`make -j1 objdir/resource.res` should fail in both case

**Comment by Carluccio** [comment]
Example files
[Moltiplicazione.zip](https://github.com/user-attachments/files/17717487/Moltiplicazione.zip)

**Reply by Carluccio** [reply]
[premake5(.lua).zip](https://github.com/user-attachments/files/17717663/premake5.lua.zip)

**Comment by Carluccio** [comment]
In premake5-beta3, the defect I mentioned only exists if the .rc file is the first in alphabetical order to the others.
